src/lib.rs

```diff
@@ -97,16 +97,64 @@
 
 mod collector;
 
+use std::{mem, sync::Arc};
+
 use metrics::{describe_gauge, gauge, Unit};
 
+/// Metrics names
+#[derive(Debug, PartialEq, Eq)]
+struct Metrics {
+    cpu_seconds_total: Arc<str>,
+    open_fds: Arc<str>,
+    max_fds: Arc<str>,
+    virtual_memory_bytes: Arc<str>,
+    virtual_memory_max_bytes: Arc<str>,
+    resident_memory_bytes: Arc<str>,
+    start_time_seconds: Arc<str>,
+    threads: Arc<str>,
+}
+
+impl Metrics {
+    // Create new Metrics, allocating prefixed strings for metrics names.
+    fn new(prefix: impl AsRef<str>) -> Arc<Self> {
+        let prefix = prefix.as_ref();
+        Arc::new(Self {
+            cpu_seconds_total: format!("{prefix}process_cpu_seconds_total").into(),
+            open_fds: format!("{prefix}process_open_fds").into(),
+            max_fds: format!("{prefix}process_max_fds").into(),
+            virtual_memory_bytes: format!("{prefix}process_virtual_memory_bytes").into(),
+            virtual_memory_max_bytes: format!("{prefix}process_virtual_memory_max_bytes").into(),
+            resident_memory_bytes: format!("{prefix}process_resident_memory_bytes").into(),
+            start_time_seconds: format!("{prefix}process_start_time_seconds").into(),
+            threads: format!("{prefix}process_threads").into(),
+        })
+    }
+}
+
+impl Default for Metrics {
+    // Create new Metrics, without prefixing and thus allocating.
+    fn default() -> Self {
+        Self {
+            cpu_seconds_total: "process_cpu_seconds_total".into(),
+            open_fds: "process_open_fds".into(),
+            max_fds: "process_max_fds".into(),
+            virtual_memory_bytes: "process_virtual_memory_bytes".into(),
+            virtual_memory_max_bytes: "process_virtual_memory_max_bytes".into(),
+            resident_memory_bytes: "process_resident_memory_bytes".into(),
+            start_time_seconds: "process_start_time_seconds".into(),
+            threads: "process_threads".into(),
+        }
```

Comment:
Is there any reason not to use `Self::new("")` here? We'd like to minimize the use of string representations such as `"process_cpu_seconds_total"` for code maintenance.

Reply:
~~`Self::new("")` will always allocate the string (due to a `format!`, albeit the no changes with the empty prefix), but `Self::default()` uses string literals~~ Nevermind, `Self::default()` allocates too, making `Arc<str>` out of `&'static str`s. I agree, will change.